src/status.rs

```diff
@@ -337,12 +346,54 @@ fn is_disconnected(channel: &Channel, messages: &Messages) -> bool {
     !(messages.has_recent_leader_hb_from(follower) && messages.has_recent_follower_hb_from(leader))
 }
 
-fn is_rejected_state() -> bool {
-    todo!()
+async fn is_rejected_state(channel: &Channel, messages: &Messages, sentry: &SentryApi) -> Result<bool, Error> {
+    let validator_messages = sentry
+        .get_latest_new_state(channel.spec.validators.leader())
+        .await?
+        .validator_messages;
+
+    let latest_new_state = validator_messages
+        .get(0)
```

Comment:
Can you move this to the `get_latest_new_state`? Since we only fetch a single Message anyway returning an `Option<NewStateValidatorMessage>` should be more clear.